test/integration-tests/IntegrationTests.Helpers/TestHelper.cs

```diff
@@ -82,9 +83,27 @@ public Container StartContainer(int traceAgentPort, int webPort)
 
         PowershellHelper.RunCommand($"docker exec {container.Name} curl -v {healthCheckEndpoint}", Output);
 
-        var (standardOutput, _) = PowershellHelper.RunCommand($"Write-Host (Test-NetConnection -ComputerName 'localhost' -Port {webPort}).TcpTestSucceeded", Output);
+        var healthChecks = 0;
+        var healthChecksUrl = $"http://localhost:{webPort}/health-check";
+        var healthChecksPassed = false;
 
-        standardOutput.Should().Contain("True", $"Test connection to the port number {webPort} failed. On this port container service should be available.");
+        do
+        {
+            var (standardOutput, _) = PowershellHelper.RunCommand($"Write-Host (Invoke-WebRequest -Uri {healthChecksUrl}).StatusCode", Output);
+
+            healthChecksPassed = standardOutput.Trim() == "200";
+
+            if (!healthChecksPassed)
+            {
+                healthChecks++;
+
+                // wait for next health check
+                Thread.Sleep(2000 * healthChecks);
```

Comment:
For tests we are not typically looking at exponential back-off: we want the test to proceed as soon as it is ready - of course, we don't want to hog resources while doing that but this should be already an I/O call so this probably is not actually needed.

Reply:
It might be possible that the non-busy wait from the timeout will be good enough, and allow the test app enough cpu time to finish starting up.